src/test/java/org/jenkinsci/main/modules/instance_identity/InstanceIdentityTest.java

```diff
@@ -48,11 +53,30 @@
         assertIdentity();
     }
 
+    @LocalData
+    @Test public void identityKeyCorrupted() throws Exception {
+        String md5CurruptedIdentityKeyFile = "527cb65e89275095a238256b3d31b07a";
+        File d = r.jenkins.getRootDir();
+        File identityKeyFile = new File(d, "identity.key.enc");
+
+        FileInputStream fis = new FileInputStream(identityKeyFile);
+        String md5NewIdentityKeyFile = org.apache.commons.codec.digest.DigestUtils.md5Hex(fis);
+        fis.close();
+        assertFalse(md5NewIdentityKeyFile.equals(md5CurruptedIdentityKeyFile));
+    }
```

Comment:
on top of this, you may want to check something like
`assertNotNull(getFileReader(md5NewIdentityKeyFile))`